iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp

```diff
@@ -21,6 +21,13 @@ namespace Flow {
 
 namespace {
 
+struct ConvertMatmulToMmt4DTargetInfo {
+  // Indicates that the target ISA is Aarch64
+  bool aarch64 = false;
```

Comment:
Can we make this a string/enum for which arch it is rather than a bool? I guess easy to change later. Same below that it would be good to make it a list of cpu features

Reply:
This mirrors the pass options, so i'm looking at that level. I can see that there is `ListOption` but:
  - I don't know how to make an enum option, short of making it a general string option and implementing c++ logic to convert that to my enum (and then it's still entirely up to me to keep the docstring in sync).
  - A ListOption, being a list, is ordered and can have duplicates. Both traits are undesirable here, what we really mean here is an 'unordered set', at least for the cpu features (I agree that the arch is different and really 'should' be an enum, but see above).

The current bag-of-named-booleans is easy, gets auto-generated command-line help that's in sync, is inherently an unordered set...

Reply:
Had a change of heart and followed your general advice on this topic now :-)